### ISSUE TITLE
multiple mini roots

### ISSUE THREAD

**Issue by yury** [holzschu/ios_system#51]
Hi,

Originals of this request:

I wanted to link `iCloud Drive/Documents` to `~/Documents/iCloud`. My first try was with hard links, but turns out iCloud do not refresh live in that case. Soft link is worked, but if I `cd ~/iCloud` I got permission denied if I set miniroot to `app/Documents`.

Proposal:

Add `ios_setMiniRoots(NSArray<NSString *> *paths)` and use collection of paths to check.

Changing miniroot is not very good option.

So, if you don't mind, I can create PR with that.

**Comment by holzschu** [comment]
Hi @yury, 
It makes a lot of sense, yes. There are already multiple miniroots: each session has its own miniroot, plus the application miniroot (we check both for operations). This allows us to open a session inside a directory in another app sandbox, and to go back to the main directory (but a session can not change into the directory of another session, even though the permissions are probably there). 

What you have described would be more powerful, so I'm all for it. 

There are practical questions: 
- how to describe the different directories? Right now, `cd` and `cd ~` go to the miniroot. If there are several miniroots, one of them (probably the first) has to be the "main" home dir. How to go to the other one? `cd ~iCloud`? If the "main" directory is the iCloud Drive Documents directory, what would be the other? `cd ~local`? 
- should we allow access to directories opened by other sessions in the same app? It would be useful. How to designate them? `cd ~2`to go to the miniroot of session 2? 
- short version: for each registered miniroot, we probably need a name to designate it. Which requires changing the tilde conversion part of `ios_system.m`(lines 206 and following). 
- should that be a fixed functionality (access only to `app/Documents`and to `iCloud Drive/Documents`) or a more general functionality?

**Comment by yury** [comment]
Hmm, for me, ideal solution would be to scan symbolic links at top level of miniroot and grant navigating to them.

But for the start, `cd`  and `cd ~` would always lead to first path from session miniroots. This behavior would not confuse users and introduce new syntax and behavior. For example:

`cp ~/iCloud/bla.txt ~/localdir/bla.txt` - will copy from `iCloudDrive/Documents/bla.txt` to `app/Documents/bla.txt`

**Comment by yury** [comment]
I’m happy with merge. Closing.
